devices/xiaomi-beryllium/default.nix

```diff
@@ -0,0 +1,25 @@
+{ config, lib, pkgs, ... }:
+
+{
+  imports = [
+    ../families/sdm845-mainline
+  ];
+
+  mobile.device.name = "xiaomi-beryllium-tianma";
```

Comment:
This should be the same name as the device folder name, which also should be what other projects use, e.g. `beryllium` for LineageOS is `xiaomi-beryllium` for us.

Though as I've now searched a bit, I see there is no automatic handling of the correct dtb if both dtb builds are selected. I do not know how this should be modeled in Mobile NixOS.

Having two distinct-but-the-same devices is kinda bad. Adding a "quirk" might be the lesser of evils, waiting for maybe some work upstream to make this work without having the build differ.

This may fly over your head, as I'm mostly pondering about details of how this is implemented, do ask if you have questions.

Reply:
Note that when using `mobile.device.name = "xiaomi-beryllium";` you'll have to do this:

https://github.com/NixOS/mobile-nixos/blob/5ee45cc1f8e43f4af14ee17ccef9156b0db8cd77/devices/families/sdm845-mainline/default.nix#L48-L50

```nix
{
  mobile.system.android = {
    appendDTB = lib.mkDefault [
      "dtbs/qcom/sdm845-${config.mobile.device.name}-tianma.dtb"
    ];
  };
}
```

And if a quirk is added, `dtbs/qcom/sdm845-${config.mobile.device.name}-${...quirk.devices.beryllium.panelVendor}.dtb`

Reply:
I understand the "concern". I've actually added this as a comment (with much less concerns and or background info) too.

I've found similar issues in for example the postmarket OS project trying to support the POCO F1.

When I tried to boot the EBBG variant (as I did not know which one I had) it booted sometimes and then would get till some state untill there was a sort of crash in libinput as far as I could tell (sorry for being vague...).

Reply:
I haven't managed to get this working as a parameter or something like that. My general nix knowledge isn't that great.